kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -236,17 +236,19 @@ public static void verifyDeltaVersions(
     }
     expectedStartVersion.ifPresent(
         v -> {
-          checkArgument(
-              !versions.isEmpty() && Objects.equals(versions.get(0), v),
-              "Did not get the first delta file version %s to compute Snapshot",
-              v);
+          if (versions.isEmpty() || !Objects.equals(versions.get(0), v)) {
+            throw new InvalidTableException(
+                tablePath.toString(),
+                String.format("Cannot compute snapshot. Missing delta file version %d.", v));
+          }
         });
     expectedEndVersion.ifPresent(
         v -> {
-          checkArgument(
-              !versions.isEmpty() && Objects.equals(ListUtils.getLast(versions), v),
-              "Did not get the last delta file version %s to compute Snapshot",
-              v);
+          if (versions.isEmpty() || !Objects.equals(ListUtils.getLast(versions), v)) {
+            throw new InvalidTableException(
+                tablePath.toString(),
+                String.format("Cannot compute snapshot. Missing delta file version %d.", v));
+          }
```

Comment:
I found this logic somewhere a little confusing, because I think we don't actually hit this right?

We calculate `newVersion` using `deltasAfterCheckpoint` and then we provide it as the endVersion to this method. I feel like maybe `IllegalArgumentException` is the more appropriate error type here?

Can you explain to me what kind of invalid table might hit this exception?

Reply:
> because I think we don't actually hit this right?

That's right. This is here as a redundancy. I'll use IllegalArgumentException !

Reply:
Synced offline. Will implement the desired approach!